### PR TITLE
Add user permission deactivation email

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Adds an email that notifies the users that their permissions are about to be deactivated
- Adds a background task for sending the deactivation email

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3598](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3598)

[TILA-3598]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3598?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tilavarauspalvelu/models/user/queryset.py [hunk 1]
```diff
@@ -21,12 +21,37 @@ def remove_old_superuser_and_staff_permissions(self) -> None:
...
             models.Q(is_active=True)
-            & models.Q(last_login__date__lt=cutoff)
             & (
                 models.Q(is_staff=True)  #
                 | models.Q(is_superuser=True)
             )
+            & models.Q(last_login__date__lt=cutoff)
         ).update(is_staff=False, is_superuser=False)
 
+    def should_deactivate_permissions(self) -> models.QuerySet:
```
Comment (vergama) [inline tilavarauspalvelu/models/user/queryset.py:31]: Wasn't explicitly specified in the ticket, but if the task is run daily then several emails would be sent to the same user, until either the permissions have eventually been deactivated or the user has logged in again -- and for the period of time (in production environment) between email and deactivation, the amount would be a bit excessive.
I guess need to store on the user model that this notification has been sent, and clear it on login.

**Review by vergama — CHANGES_REQUESTED** [review]
The email texts look good, but see comment about limiting the email to be sent just once prior to the upcoming deactivation.

File: tilavarauspalvelu/models/user/queryset.py (hunk 1 above)
Comment (matti-lamppu) [reply tilavarauspalvelu/models/user/queryset.py:31]: Added field `sent_email_about_deactivating_permissions` to user and then checks for this field when gathering email recipients for the email, as well as an update to the after the emails have been sent.

**Review by vergama — APPROVED** [review]

**Pull request merged**